keps/sig-apps/2255-pod-cost/README.md

```diff
@@ -55,163 +54,183 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+One risk is that users perceive the feature as a guarantee to delete order. Documentation
+should stress the fact that this is best effort.
 
 ## Design Details
 
+The pod-deletion-cost range will be from [-MaxInt, MaxInt]. The default value is 0.
+Invalid values (like setting the annotation to string) will be converted to
+the default value.
+
+Having the default value in the middle of the range allows controllers to cutomize
+the semantics of the cost of deleting pods that don't have the annotation set: 
+controllers can use positive pod-deletion-cost values if they always want uninitialized
+pods to be deleted first, or use negative pod-deletion-cost values if they want
+uninitialized pods to always be deleted last.
+
+When scaling down a ReplicaSet, controller-manager will prioritize deleting
+pods with lower pod-deletion-cost. Specifically, the pod-deletion-cost will be evaluated after
+step 3 and before step 4 as they are currently defined in
+[ActivePodsWithRanks](https://github.com/kubernetes/kubernetes/blob/cac933934b1301665e6e51a81c66c483f4e16c49/pkg/controller/controller_utils.go#L784-L809),
+which means the followig criteria is applied when comparing two pods regardless of their pod-deletion-cost:
+- if one is assigned a node and the other is not, then the unassigned pod is deleted first.
+- if the two pods are in different phases, then the pod in pending/unknown status is deleted first.
+- if the two pods have different readiness status, then the not ready pod is deleted first
+
+
+If none of the pods set the pod-deletion-cost annotation or all of them have the same value, then the 
+scale down behavior is not changed compared to now.
 
 ### Test Plan
 
-* Units test in kube-controller-manager package to test a variety of scenarios.
-* New E2E Tests to validate that replicas get deleted as expected e.g:
- * Replicas with lower pod-cost before replicas with higher pod-cost
- * Replicas with no pod-cost annotation set before replicas with low priority
+- Units test in kube-controller-manager package to test a variety of scenarios.
+- Integration tests to validate that:
+  - Replicas with lower pod-deletion-cost are deleted before replicas with higher pod-deletion-cost
+  - No behavior change when pod-deletion-cost is not set or all pods have the same pod-deletion-cost
 
 ### Graduation Criteria
 
 #### Alpha -> Beta Graduation
 * Implemented feedback from alpha testers
```

Comment:
I see that you removed "monitoring" section from the PRR. It's fine for Alpha, but please add a point that metrics are agreed on and implemented for Beta graduation criteria

Reply:
I updated the monitoring section apart from the two questions related to SLI and SLO, I am not sure if there is a practical way of computing an SLI/SLO for this feature.

Reply:
I see no answers in monitoring section ...

Reply:
I updated them to N/A since I don't think there is a practical way of computing SLO/SLI for such a feature.